src/triage/predictlist/__init__.py

```diff
@@ -0,0 +1,610 @@
+        }
+
+        last_split_train_matrix_uuid, last_split_matrix_metadata = train_matrix_info_from_model_id(
+            self.db_engine, 
+            model_id=self.model_group_info['model_id_last_split']
+        )
+
+        random_seed = self.model_trainer.get_or_generate_random_seed( 
+            model_group_id=self.model_group_id, 
+            matrix_metadata=last_split_matrix_metadata, 
+            train_matrix_uuid=last_split_train_matrix_uuid
+        )
+
+        retrained_model_id = self.model_trainer.process_train_task(
+            matrix_store=self.matrix_storage_engine.get_store(matrix_uuid), 
+            class_path=self.model_group_info['model_type'], 
+            parameters=self.model_group_info['hyperparameters'], 
+            model_hash=None, 
+            misc_db_parameters=misc_db_parameters, 
+            random_seed=random_seed, 
+            retrain=True,
+            model_group_id=self.model_group_id
+        )
+
+        self.retrained_model_hash = retrieve_model_hash_from_id(self.db_engine, retrained_model_id)
+        self.retrained_matrix_uuid = matrix_uuid
+        self.retrained_model_id = retrained_model_id
+        
+        return {'retrain_model_comment': retrain_model_comment}
+
+    def predict(self, prediction_date):
+        """Predict forward by creating a matrix using as_of_date = prediction_date and applying the retrained model on it
+
+        Args:
+            prediction_date(str)
+        """
+        cohort_table_name = f"triage_production.cohort_{self.experiment_config['cohort_config']['name']}_predict"
+
+        # 1. Generate cohort
+        self.generate_entity_date_table(prediction_date, cohort_table_name)
+
+        # 2. Generate feature aggregations
+        collate_aggregations = self.get_collate_aggregations(prediction_date, cohort_table_name)
+        self.feature_generator.process_table_tasks(
+            self.feature_generator.generate_all_table_tasks(
+                collate_aggregations,
+                task_type='aggregation'
+            )
+        )
+        # 3. Reconstruct feature disctionary from feature_names and generate imputation
+        reconstructed_feature_dict, imputation_table_tasks = self.get_feature_dict_and_imputation_task(
+                collate_aggregations, 
+                self.retrained_model_id
+        )
+        self.feature_generator.process_table_tasks(imputation_table_tasks)
+ 
+        # 4. Build matrix
+        db_config = {
+            "features_schema_name": "triage_production",
+            "labels_schema_name": "public",
+            "cohort_table_name": cohort_table_name,
+        }
+
+        matrix_builder = MatrixBuilder(
+            db_config=db_config,
+            matrix_storage_engine=self.matrix_storage_engine,
+            engine=self.db_engine,
+            experiment_hash=None,
+            replace=True,
+        )
+        # Use timechop to get the time definition for production
```

Comment:
up above it looks like you generate the cohort table just for the `prediction_date` as a single `as_of_date` but then here you'll potentially end up with multiple "test" `as_of_date`s from `Timechop` right?